scarlet/model.py

```diff
@@ -66,14 +68,34 @@ def __next__(self):
         return self._children.__next__()
 
     def get_parameter(self, i, *parameters):
+        """Access parameters by list index or by name
+
+        Parameters
+        ----------
+        i: int, slice, str
+            Index, slice or name attribute of the requested parameter
+        parameters: tuple
+            Parameters used during optimization. If not set, uses `self`
+
+        Returns
+        -------
+        Matching item or tuple of matching items
+        """
+
         # NOTE: index lookup only works if order is not changed by parameter fixing!
-        # check parameters first during optimization
-        if i < len(parameters):
-            return parameters[i]
+        # during optimization: parameters are passed by autograd
+        if parameters:
+            parameters_ = parameters
+        else:
+            parameters_ = self.parameters
 
-        # otherwise use self
-        if i < len(self.parameters):
-            return self.parameters[i]
```

Comment:
Was this actually a bug? `parameters` was only used when it was longer than `self.parameters`?

Reply:
No. `parameters` is only an argument to the function during optimization, otherwise it accesses them through `self`. They are identical arrays, but the former has an `autograd` box around it.

Reply:
Glad I asked, it seems like it poses a problem in the new version (I'm pulling the psf_shift, but it says `parameters` is an `ArrayBox`)
```~/Desktop/git_repos/scarlet/scarlet/model.py in get_parameter(self, i, *parameters)
     95             for p in parameters_:
     96                 print(p)
---> 97                 print(p.name)
     98             match = tuple(p for p in parameters_ if p.name == i)
     99             if len(match) == 1:

AttributeError: 'ArrayBox' object has no attribute 'name'```

Reply:
good catch! Let me fix that